protobuf_serialization/files/type_generator.nim

```diff
@@ -64,73 +138,53 @@ proc protoToTypesInternal*(filepath: string): NimNode =
-            of "double":
-              value[2][^1][0][1].add(ident("pfloat64"))
-              value[2][^1][1] = ident("float32")
-            of "float32":
-              value[2][^1][0][1].add(ident("pfloat32"))
-            of "int32":
-              value[2][^1][0][1].add(ident("pint"))
-            of "int64":
-              value[2][^1][0][1].add(ident("pint"))
-            of "uint32":
-              value[2][^1][0][1].add(ident("pint"))
-            of "uint64":
-              value[2][^1][0][1].add(ident("pint"))
-            of "sint32":
-              value[2][^1][0][1].add(ident("sint"))
-              value[2][^1][1] = ident("int32")
-            of "sint64":
-              value[2][^1][0][1].add(ident("sint"))
-              value[2][^1][1] = ident("int64")
-            of "fixed32":
-              value[2][^1][0][1].add(ident("fixed"))
-              value[2][^1][1] = ident("uint32")
-            of "fixed64":
-              value[2][^1][0][1].add(ident("fixed"))
-              value[2][^1][1] = ident("uint64")
-            of "sfixed32":
-              value[2][^1][0][1].add(ident("fixed"))
-              value[2][^1][1] = ident("int32")
-            of "sfixed64":
-              value[2][^1][0][1].add(ident("fixed"))
-              value[2][^1][1] = ident("int64")
-            of "bool":
-              discard
-            of "string":
-              discard
-            of "bytes":
-              repeated += 1
-              value[2][^1][1] = ident("byte")
+          if value[2][^1][1].strVal.startsWith("map<"):
+            var matches = value[2][^1][1].strVal.split({'<', '>', ','})
+            let (typ1, _) = getTypeAndPragma(matches[1])
+            let (typ2, _) = getTypeAndPragma(matches[2])
+            value[2][^1][1] = nnkBracketExpr.newTree(newIdentNode("Table"), typ1, typ2)
+          else:
+            let (typ, pragma) = getTypeAndPragma(value[2][^1][1].strVal)
+            value[2][^1][1] = typ
+            if not pragma.isNil():
+              value[2][^1][0][1].add(pragma)
 
-          for _ in 0 ..< repeated:
+          if field.presence == Repeated:
             value[2][^1][1] = newNimNode(nnkBracketExpr).add(
               ident("seq"),
               value[2][^1][1]
             )
-
+          elif isReference:
+            value[2][^1][1] = newNimNode(nnkRefTy).add(value[2][^1][1])
         if value[2].len == 0:
           value[2] = newEmptyNode()
 
+
       result.add(
         newNimNode(nnkTypeDef).add(
           newNimNode(nnkPragmaExpr).add(
             newNimNode(nnkPostfix).add(ident("*"), ident(name)),
-            newNimNode(nnkPragma).add(ident("proto3"))
+            if next.kind == ProtoType.Enum:
+              newNimNode(nnkPragma).add(ident("pure"), ident("proto3"))
+            else:
+              newNimNode(nnkPragma).add(ident("proto3"))
           ),
           newEmptyNode(),
           value
         )
       )
+  if logFile != "":
+    logFile.writeFile(repr(result))
```

Comment:
Instead of this, you can try to use the stew system:
https://github.com/status-im/nim-stew/blob/c52569abd5a7753c1892bf9fc85c71f0c9b6ac45/stew/shims/macros.nim#L57

`result.storeMacroResult()`
And then the caller can do `writeMacroResultsNow` if he wants to